scipy/signal/ltisys.py

```diff
@@ -120,20 +120,38 @@ def tf2ss(num, den):
 
     if num.shape[-1] > 0:
         D = num[:, 0]
+
+        # This will occur if 'num' is 2-D, so
+        # we need to ensure that D itself is 2-D
+        if len(D.shape) == 1:
+            D = asarray([D], D.dtype)
+
     else:
         # We don't assign it an empty array because this system
         # is not 'null'. It just doesn't have a non-zero D
         # matrix. Thus, it should have a non-zero shape so that
         # it can be operated on by functions like 'ss2tf'
-        D = array([0], float)
+        D = array([[0]], float)
 
     if K == 1:
-        return array([0], float), array([0], float), array([0], float), D
+        D = D.reshape(num.shape)
+
+        return (zeros((1, 1)), zeros((1, D.shape[1])),
+                zeros((D.shape[0], 1)), D)
 
     frow = -array([den[1:]])
     A = r_[frow, eye(K - 2, K - 1)]
     B = eye(K - 1, 1)
-    C = num[:, 1:] - num[:, 0] * den[1:]
+
+    left = num[:, 0]
+    left = left.reshape((len(left), 1))
+
+    right = den[1:]
+    right = right.reshape((1, len(right)))
+
+    C = num[:, 1:] - left * right
```

Comment:
I think this might be the same as doing `np.outer(num[:, 0], den[1:])`? If so, the latter might be clearer.